cocos/scripting/js-bindings/manual/jsb_global.cpp

```diff
@@ -4,27 +4,83 @@
 
 #include "jsb_global.h"
 #include "jsb_conversions.hpp"
+#include "xxtea/xxtea.h"
 
 using namespace cocos2d;
 
 se::Object* __jscObj = nullptr;
 se::Object* __ccObj = nullptr;
 se::Object* __jsbObj = nullptr;
 
+const char* BYTE_CODE_FILE_EXT = ".jsc";
+
+static std::string xxteaKey = "";
+
+void jsb_set_xxtea_key(const std::string& key)
+{
+    xxteaKey = key;
+}
+
+static std::string removeFileExt(const std::string& filePath)
+{
+    size_t pos = filePath.rfind('.');
+    if (0 < pos)
+    {
+        return filePath.substr(0, pos);
+    }
+    return filePath;
+}
+
 bool jsb_run_script(const std::string& filePath)
 {
     static se::ScriptEngine::FileOperationDelegate delegate;
     if (!delegate.isValid())
     {
         delegate.onGetDataFromFile = [](const std::string& path, const uint8_t** data, size_t* dataLen) -> void{
             assert(!path.empty() && data != nullptr && dataLen != nullptr);
-            Data fileData = FileUtils::getInstance()->getDataFromFile(path);
+            
+            Data fileData;
+            
+#if SCRIPT_ENGINE_TYPE != SCRIPT_ENGINE_SM
+            std::string byteCodePath = removeFileExt(path) + BYTE_CODE_FILE_EXT;
+            if (FileUtils::getInstance()->isFileExist(byteCodePath)) {
+                fileData = FileUtils::getInstance()->getDataFromFile(byteCodePath);
+                
+                *data = xxtea_decrypt((unsigned char*)fileData.getBytes(), (uint32_t)fileData.getSize(), (unsigned char*)xxteaKey.c_str(), (uint32_t)xxteaKey.size(), (uint32_t*)&dataLen);
```

Comment:
It's will cause memory leak here.
I looked at the code of `xxtea_decrypt`, found that it returns a pointer by `malloc`, therefore, we need to use `free` to release the memory.

`*data = xxtea_decrypt(...)`  uses the `Data` copy assignment which will copy the data rather than take the ownership. so here it will leak.